Bottom: 19271e553f8356e277d89e7290eed1fd687e74fa
Top:    86c891af5fc2a9d159e5295e4828dccf64512894
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-10 15:49:03 +0100

Refresh of playing-with-stack-rendering

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e3652dd..e5bfd30 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1079,8 +1079,8 @@ SetPVColor_inlayer (PinTypePtr Pin, LayerTypePtr Layer, int Type)
     color = PCB->ConnectedColor;
   else
     {
-      int component_group = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
-      int solder_group    = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
+      int component_group = GetLayerGroupNumberByNumber (component_silk_layer);
+      int solder_group    = GetLayerGroupNumberByNumber (solder_silk_layer);
       int this_group      = GetLayerGroupNumberByPointer (Layer);
 
       if (this_group == component_group || this_group == solder_group)
@@ -1374,8 +1374,8 @@ DrawLayerGroup (int group, const BoxType * screen)
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
   int first_run = 1;
-  int component = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
-  int solder    = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
+  int component = GetLayerGroupNumberByNumber (component_silk_layer);
+  int solder    = GetLayerGroupNumberByNumber (solder_silk_layer);
 
   if (!gui->set_layer (0, group, 0)) {
     gui->set_layer (NULL, SL (FINISHED, 0), 0);
